### PR TITLE
gh#12397: introduce iban settings for EU countries

### PR DESCRIPTION
**Pull request by Chemssou**

### CODE REVIEW

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+update c_country  set c_currency_id  = 102 
+                      , bankcodelength      = 4 	   , bankcodechartype      = 'n'			, bankcodeseqno      = 10
+                      , branchcodelength    = null 	   , branchcodechartype    = null		    , branchcodeseqno    = null
+                      , accountnumberlength = 16 	   , accountnumberchartype = 'n' 			, accountnumberseqno = 20                                             
+                      , nationalcheckdigitlength=null    , nationalcheckdigitchartype = null	, nationalcheckdigitseqno  = null
+where countrycode  =  'SK'
+; 
+
+-- RO: Romania
+update c_country  set c_currency_id  = 102
```
Comment (teosarca) [inline backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql:173]: Romania, unfortunately does not have EUR but RON (346)

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+update c_country  set c_currency_id  = 102 
+                      , bankcodelength  = 4		 	   , bankcodechartype ='n'					, bankcodeseqno  = 10
+                      , branchcodelength = 4	 	   , branchcodechartype = 'n'				, branchcodeseqno  = 20
+                      , accountnumberlength = 12 	   , accountnumberchartype ='c' 			, accountnumberseqno =30                                             
+                      , nationalcheckdigitlength=null  , nationalcheckdigitchartype = null		, nationalcheckdigitseqno  = null
+where countrycode  =  'AD'
+;  
+
+-- CZ
+update c_country  set c_currency_id  = 102
```
Comment (teosarca) [inline backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql:18]: Czech koruna

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+update c_country  set c_currency_id  = 102 
+                      , bankcodelength  = 4		 	, bankcodechartype ='n'					, bankcodeseqno  = 10
+                      , accountnumberlength = 10 	, accountnumberchartype ='n' 			, accountnumberseqno =20                       
+                      , branchcodelength = null	 	, branchcodechartype = null				, branchcodeseqno  = null
+                      , nationalcheckdigitlength = null, nationalcheckdigitchartype = null	, nationalcheckdigitseqno = null
+where countrycode  =  'CZ'
+;
+
+-- HR: Croatia
+update c_country  set c_currency_id  = 102
```
Comment (teosarca) [inline backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql:27]: kuna

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+update c_country  set c_currency_id  = 102 
+                      , bankcodelength      = 6 	   , bankcodechartype      = 'n'			, bankcodeseqno      = 10
+                      , branchcodelength    = null	   , branchcodechartype    = null		    , branchcodeseqno    = null
+                      , accountnumberlength = 7 	   , accountnumberchartype = 'n' 			, accountnumberseqno = 20                                             
+                      , nationalcheckdigitlength=1     , nationalcheckdigitchartype = 'n'		, nationalcheckdigitseqno  = 30
+where countrycode  =  'FI'
+; 
+
+-- HU: Hungary
+update c_country  set c_currency_id  = 102
```
Comment (teosarca) [inline backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql:63]: forint

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+update c_country  set c_currency_id  = 102 
+                      , bankcodelength      = 5 	   , bankcodechartype      = 'n'			, bankcodeseqno      = 10
+                      , branchcodelength    = null	   , branchcodechartype    = null		    , branchcodeseqno    = null
+                      , accountnumberlength = 11 	   , accountnumberchartype = 'n' 			, accountnumberseqno = 20                                             
+                      , nationalcheckdigitlength=null     , nationalcheckdigitchartype = null	, nationalcheckdigitseqno  = null
+where countrycode  =  'LT'
+; 
+
+-- PL: Poland
+update c_country  set c_currency_id  = 102
```
Comment (teosarca) [inline backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql:81]: zlot

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+update c_country  set c_currency_id  = 102 
+                      , bankcodelength      = 2 	   , bankcodechartype      = 'n'			, bankcodeseqno      = 10
+                      , branchcodelength    = 3 	   , branchcodechartype    = 'n'		    , branchcodeseqno    = 20
+                      , accountnumberlength = 8 	   , accountnumberchartype = 'n' 			, accountnumberseqno = 30                                             
+                      , nationalcheckdigitlength=2     , nationalcheckdigitchartype = 'n'		, nationalcheckdigitseqno  = 40
+where countrycode  =  'SI'
+; 
+
+-- SE: Sweden
+update c_country  set c_currency_id  = 102
```
Comment (teosarca) [inline backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql:108]: Swedish krona

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+update c_country  set c_currency_id  = 102 
+                      , bankcodelength      = 5 	   , bankcodechartype      = 'n'			, bankcodeseqno      = 10
+                      , branchcodelength    = 5 	   , branchcodechartype    = 'n'		    , branchcodeseqno    = 20
+                      , accountnumberlength = 11 	   , accountnumberchartype = 'c' 			, accountnumberseqno = 30                                             
+                      , nationalcheckdigitlength=2    , nationalcheckdigitchartype = 'n'	, nationalcheckdigitseqno  = 40
+where countrycode  =  'MC'
+; 
+
+-- BG: Bulgaria
+update c_country  set c_currency_id  = 102
```
Comment (teosarca) [inline backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5623040_sys_gh12397_introduce_iban_settings.sql:135]: leva

**Comment by teosarca** [comment]
@Chemssou  only the countries of European Monetary Union have EURO currency. see https://en.wikipedia.org/wiki/Eurozone

**Comment by teosarca** [comment]
or to better understand the supranational unions from Europe: https://commons.wikimedia.org/wiki/File:Supranational_European_Bodies-en.png

hint: the eurozone is for EUR currency

**Review by metas-ts — APPROVED** [review]

**Pull request merged**